sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/Uri.java

```diff
@@ -54,12 +58,14 @@ public String getURIAsString() {
         return this.uriAsString;
     }
 
-    /***
-     * Attention: This is only used for fault injection to easier decide whether the address is primary address.
-     * @param primary
-     */
     public void setPrimary(boolean primary) {
         isPrimary = primary;
+        if (primary) {
+            this.healthStatusTuple.updateAndGet(previousStatusTuple -> {
```

Comment:
Can't we re-instantiate instead of replace and keep `HealthStatusAndDiagnosticStringTuple` immutable?

Reply:
Changed it to reinstantiate.